### PR TITLE
feat: using slice to be faster

### PR DESCRIPTION
**Pull request by azl397985856**
call the method insert on the list a, whereas all the operations in slice are handled directly in byte code.

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@azl397985856

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by tirkarthi** [comment]
The code changes should be discussed in https://bugs.python.org. It also causes test failures. cc: @rhettinger

File: Lib/bisect.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def insort_right(a, x, lo=0, hi=None):
         mid = (lo+hi)//2
         if x < a[mid]: hi = mid
         else: lo = mid+1
-    a.insert(lo, x)
+    a[lo:lo] = x
```
Comment (da-woods) [inline Lib/bisect.py:20]: They aren't the same though. `a.insert(lo, x)` inserts an item into `a`. `a[lo:lo]=x` inserts all the items of an iterable into `a` (and fails if `x` isn't iterable).

Did you test this?

Comment (azl397985856) [reply Lib/bisect.py:20]: ok~ seems reasonable

**Pull request closed without merging**